okpt/io/config/parsers/nmslib.py

```diff
@@ -19,12 +19,34 @@
 Classes:
     NmslibParser: NMSLIB config parser.
 """
+from dataclasses import dataclass
 from io import TextIOWrapper
-from typing import Any, Dict
 
 from okpt.io.config.parsers import base
 
 
+@dataclass
+class MethodParametersConfig:
+    ef_construction: int
+    ef_search: int
+    post: int
+    m: int
+
+
+@dataclass
+class MethodConfig:
```

Comment:
Could space_type be considered a parameter? If so, is there any need for this class?

Reply:
It could, but I left `space_type` out of `parameters` to parallel the OpenSearch index settings format. I've also left a note in the dataclass explaining this reasoning.